bots/machine/testvm_core.py

```diff
@@ -0,0 +1,557 @@
+#!/usr/bin/python3 -u
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2013 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+
+import os
+import time
+import socket
+import signal
+import subprocess
+import tempfile
+import select
+import errno
+import sys
+
+BOTS_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+TEST_DIR = os.path.join(os.path.dirname(BOTS_DIR), "test")
```

Comment:
TEST_DIR really is a layering violation here. Nothing in bots/ should know about and refer to tests/, as bots/ gets used in projects other than cockpit as well. I know you just moved it from testvm.py (where it is already wrong), so let's clean this up in a follow-up.